lib/Monads/trace/Trace_Monad.thy

```diff
@@ -645,7 +653,8 @@ subsection "Loops"
 text \<open>
   Loops are handled using the following inductive predicate;
   non-termination is represented using the failure flag of the
-  monad.\<close>
+  monad.
+FIXME: update comment about non-termination\<close>
```

Comment:
was this FIXME intended to be left in?

Reply:
Sort of, or at least it still needs to be updated. I'll either need to stare at this more to work out how non-termination is represented here or get help from someone who understands how these loops are constructed better than me.

Reply:
We should leave the FIXME in. I don't think we have properly thought about how this should be defined yet. In particular, in the concurrent setting it does make sense to have a semantics for non-terminating loops, so the whole definition here might have to be rethought (or maybe a potentially non-terminating loop should get a separate definition).